### ISSUE TITLE
couchdb clustering not working

### ISSUE THREAD

**Issue by csantanapr** [apache/openwhisk#3029]
Took a look at the couchdb clustering and attempted to fix.
I got very far but I'm getting an error which I don't know now how to fix.

```
TASK [couchdb : add remote nodes to the cluster] *********************************************************************************************************************
Wednesday 29 November 2017  15:33:30 -0500 (0:00:00.323)       0:00:36.662 ****
fatal: [db0]: FAILED! => {"cache_control": "must-revalidate", "changed": false, "connection": "close", "content": "{\"error\":\"conflict\",\"reason\":\"Document update conflict.\"}\n", "content_length": "58", "content_type": "application/json", "date": "Wed, 29 Nov 2017 20:33:31 GMT", "failed": true, "json": {"error": "conflict","reason": "Document update conflict."}, "msg": "Status code was not [201]: HTTP Error 409: Conflict", "redirected": false, "server": "CouchDB/2.1.1 (Erlang OTP/17)","status": 409, "url": "http://172.17.0.1:5984/_cluster_setup", "x_couch_request_id": "d6757efe56", "x_couchdb_body_time": "0"}

```

I think we should consider reverting the commit https://github.com/apache/incubator-openwhisk/pull/2810
until it get's properly fix.
What do you think @cbickel @style95 ?

**Comment by csantanapr** [comment]
@cbickel @style95 you can see changes here: https://github.com/apache/incubator-openwhisk/pull/3030

**Comment by style95** [comment]
I am checking on it.

**Comment by style95** [comment]
@csantanapr Are you using dns name in db section of `hosts` file?

**Comment by csantanapr** [comment]
Give it a try on your system. 
No dns use I try to use the ansible_host value

**Comment by csantanapr** [comment]
I’m using docker for Mac
Running the following command
ansible-playbook -i environments/local couchdb.yml

**Comment by style95** [comment]
I have discussed with @csantanapr.
Actually there is limitation in #2810.

With that change, we can't setup multiple couchdb nodes on a single machine.
This limitation comes from erlang epmd port.

It is briefly explained here: https://github.com/apache/incubator-openwhisk/pull/2810#issuecomment-343837080

When couchdb form a cluster it counts on erlang epmd to find other nodes.
All nodes will try to find each other with same port(`4369`).
Since it is not possible to differentiate this port for each nodes, we can't run multiple nodes on a single machine.

One idea is, make all nodes communicate with each other using container IP.
If they are on same machine, they will be in the same docker network and reachable to each other.
But I am not sure if it works, and it requires many dirty changes.

So @cbickel and I tried to take current approach.

@csantanapr advised me to clearly write down current situation in README.
I should expect many people might be confused.
I will work on it.
